Update api-management-caching-policies.md

sunmorgus opened the pull request:
1. Set vary-by-developer and vary-by-developer-groups required column to 'Yes' (the azure portal indicated they were required)
2. Fix the name of the allow-private-response-caching comment in the Policy statement definition

azurecla commented:
Hi __@ShawnJackson__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-shawja). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

sunmorgus committed:
Update api-management-caching-policies.md

1. Set vary-by-developer and vary-by-developer-groups required column to 'Yes' (the azure portal indicated they were required)
2. Fix the name of the allow-private-response-caching comment in the Policy statement definition
